packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,172 @@
+        relay.sender.substr(2).toLowerCase() === parsedAncillaryData.l2Sender &&
+        relay.l1Token.substr(2).toLowerCase() === parsedAncillaryData.l1Token &&
+        relay.amount === parsedAncillaryData.amount.toString() &&
+        relay.slowRelayFeePct === parsedAncillaryData.slowRelayFeePct.toString() &&
+        relay.instantRelayFeePct === parsedAncillaryData.instantRelayFeePct.toString() &&
+        relay.quoteTimestamp === parsedAncillaryData.quoteTimestamp &&
+        relay.realizedLpFeePct === parsedAncillaryData.realizedLpFeePct.toString() &&
+        relay.slowRelayer.substr(2).toLowerCase() === parsedAncillaryData.slowRelayer &&
+        relay.depositContract.substr(2).toLowerCase() === parsedAncillaryData.depositContract
+    );
+
+    // TODO: Do we need to handle the case where all of these params are matched and matchedRelay.length > 1?
+    if (matchedRelay.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No relay event found for price request time",
+        priceRequestTime: time,
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    const relay = matchedRelay[0];
+
+    // Validate that relay matches a deposit struct exactly.
+    const matchedDeposits = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === relay.depositId &&
+        deposit.depositHash === relay.depositHash &&
+        deposit.timestamp === relay.depositTimestamp &&
+        deposit.sender === relay.sender &&
+        deposit.recipient === relay.recipient &&
+        deposit.l1Token === relay.l1Token &&
+        deposit.amount === relay.amount &&
+        deposit.slowRelayFeePct === relay.slowRelayFeePct &&
+        deposit.instantRelayFeePct === relay.instantRelayFeePct &&
+        deposit.quoteTimestamp === relay.quoteTimestamp
+    );
+    if (matchedDeposits.length > 1) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "TODO: Handle multiple deposits associated with same relay attempt",
+      });
+      throw new Error("TODO: Handle multiple relays for same price request timestamp");
+    } else if (matchedDeposits.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No deposit event matching relay attempt",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // Validate relays proposed realized fee percentage.
+    const expectedRealizedFeePct = this.l1Client.calculateRealizedLpFeesPctForDeposit(/* matchedDeposits[0] */);
+    if (expectedRealizedFeePct !== relay.realizedLpFeePct) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "Matched deposit with relay but realized fee % is incorrect",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // TODO: Do we need to check other parameters like slow relayer address and deposit box contract address? These
```

Comment:
Hmm interesting point. Why did we include these in ancillary data?

Reply:
So I thought to include the `depositBoxContract` address because we want users to know where to look on L2 for the `Deposit` event, and technically the `depositBoxAddress` mapped to the `BridgePool` can change after the `Deposit` has emitted. If the `depositBoxAddress` changes then how would users know which address to look up? I think this is probably over-engineering but curious what you think @chrismaree . One change we might consider making is adding the `depositBoxAddress` to the `Relay` struct stored in the `BridgePool` contract. This would be a straightforward change to make.

Regarding the `slowRelayer`, I think you can ignore this comment since we check the `slowRelayer` included in the `relayAncillaryData` against the `relayData.slowRelayer` fetched by the `l1Client`. I'll change that comment.

Reply:
Actually, looks like the `depositBoxAddress` is already emitted in the `DepositRelayed` L1 event and I already check it. So this comment can be removed.

Reply:
SGTM!